HotelBookingTests/HotelBaseClassTest.cs

```diff
@@ -32,5 +35,32 @@ public void ShouldReturnFalseIfPaymentFailed()
             Assert.That(_hotel.MakePayment(250, 200), Is.EqualTo(true));
             Assert.That(_hotel.MakePayment(250, 00), Is.EqualTo(false));
         }
+
+        [Test]
+        public void ShouldProcessPartialSteps()
+        {
+            DateTime date = DateTime.Now;
+            double price = 0;
+            int creditCardNumber = 22;
+            string email = "test";
+            PaymentService.Stub(x => x.Pay(creditCardNumber, price)).Return(true);
+
+            _hotel.Stub(action => action.Operations).Return(new List<HotelOperation>
+            {
+                new HotelOperation(Operation.CheckPrice, 1, true),
+                new HotelOperation(Operation.MakePayment, 2, true),
+                new HotelOperation(Operation.BookRoom, 3, false),
+                new HotelOperation(Operation.SendEmail, 4, true),
+                new HotelOperation(Operation.GenerateReservationNumber, 5, false)
+            });
+
+            ReservationResult result = _hotel.Reserve(date, price, creditCardNumber, email);
+            Assert.False(result.Success);
```

Comment:
Should have:

You have split this test for small ones for instance:
* ShouldIncludeSendEmailOperationResult, ShouldIncludeBookRoomOperationResult and so on
* WhenOperationShouldStopOnErrorAndSomeErrorHasOccured_ShouldNotExcuteOtherOperations
* WhenEachOperationExecutesWithoutError_ShouldPerformAllOperations - this test can look like yours, but:
  *  I will stub all operations not just payment to clearly show, from where result comes from
  * I will create expected result and compare result to result suing fluent assertions nad EquivalentTo instead of making separate asserts line here.

Happy to explain face to face if it is not clear

Reply:
Applied